pkg/database/mysql.go

```diff
@@ -0,0 +1,26 @@
+package database
+
+
+type MySQLDatabase struct {
+	DB *bun.DB
+}
+
+var (
+	once sync.Once
+	dbInstance *MySQLDatabase
+)
+
+func NewMySQLDatabase(conf *entities.Config) *MySQLDatabase {
+	once.Do(func() {
+		sqlDB, err := sql.Open("mysql", "")
+		if err != nil {
+			panic(err)
+		}
+		dbInstance := bun.NewDB(sqlDB, mysqldialect.New())
+	})
+	return dbInstance
+}
+
+func (MySQLDatabase *MySQLDatabase) GetDB() *bun.DB {
+	return dbInstance.DB
+}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Method not using receiver</b></div>
<div id="fix">

The `GetDB` method is using the package-level `dbInstance` instead of the receiver. Consider using the receiver to access the `DB` field.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -24,3 +24,3 @@
 func (MySQLDatabase *MySQLDatabase) GetDB() *bun.DB {
 -	return dbInstance.DB
 +	return MySQLDatabase.DB
 }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/productivite/pull/13#issuecomment-2437741040>#10bff6</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged